syntax = "proto3";
import "common.proto";
import "parsing_base.proto";
import "perception_base.proto";
package autodrive.workflow.proto;

enum IpmParsingLabel {
  kIpmParsingLabelRoad = 0;
  kIpmParsingLabelTrafficLine = 1;
  kIpmParsingLabelParkingLine = 2;
  kIpmParsingLabelParkingSpace = 3;
  kIpmParsingLabelTrafficArrow = 4;
  kIpmParsingLabelGuideLine = 5;
  kIpmParsingLabelCrosswalkLine = 6;
  kIpmParsingLabelNoParkingSignLine = 7;
  kIpmParsingLabelStopLine = 8;
  kIpmParsingLabelSpeedBump = 9;
  kIpmParsingLabelSignLine = 10;
  kIpmParsingLabelParkingLockOpen = 11;
  kIpmParsingLabelParkingLockClosed = 12;
  kIpmParsingLabelTrafficCone = 13;
  kIpmParsingLabelParkingRod = 14;
  kIpmParsingLabelCurb = 15;
  kIpmParsingLabelColumn = 16;
  kIpmParsingLabelImmovableObstacle = 17;
  kIpmParsingLabelMovableObstacle = 18;
  kIpmParsingLabelOther = 19;
  kIpmParsingLabelSidewalk = 20;
  kIpmParsingLabelPaintedWallRoot = 21;
}

enum FreeSpaceLabel {
  kFreeSpaceLabelBackGround = 0;
  kFreeSpaceLabelImmovableObstacle = 1;
  kFreeSpaceLabelMovableObstacle = 2;
  kFreeSpaceLabelParkingRod = 3;
}

message IpmParsingMsg {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  IpmParsingImageData free_space = 3;
  IpmParsingImageData feature_map = 4;
  // RLE compressed parsing results
  RleCompressedData rle_parsing_compressed = 5;
  // Parsing scale comparing to camera img width
  float scale = 6;
  // Valid roi on parsing img(not camera image)
  perception_base.Rect roi_output = 7;
}